src/input/TCA8418Keyboard.cpp

```diff
@@ -0,0 +1,563 @@
+    _TCA8418_COL6, // Pin ID for column 6
+    _TCA8418_COL7, // Pin ID for column 7
+    _TCA8418_COL8, // Pin ID for column 8
+    _TCA8418_COL9  // Pin ID for column 9
+};
+
+// Nokia 5130 keyboard size
+#define _TCA8418_ROWS 5
+#define _TCA8418_COLS 5
+#define _TCA8418_NUM_KEYS 16
+
+#define _TCA8418_LONG_PRESS_THRESHOLD 2000
+#define _TCA8418_MULTI_TAP_THRESHOLD 750
+
+uint8_t TCA8418TapMod[16] = {1, 1, 1, 1, 13, 7, 9, 2,
+                             7, 7, 7, 2, 7,  7, 9, 2}; // Num chars per key, Modulus for rotating through characters
+
+unsigned char TCA8418TapMap[16][13] = {{_TCA8418_BSP},                                                     // C
+                                       {_TCA8418_SELECT},                                                  // Navi
+                                       {_TCA8418_UP},                                                      // Up
+                                       {_TCA8418_DOWN},                                                    // Down
+                                       {'1', '.', ',', '?', '!', ':', ';', '-', '_', '\\', '/', '(', ')'}, // 1
+                                       {'4', 'g', 'h', 'i', 'G', 'H', 'I'},                                // 4
+                                       {'7', 'p', 'q', 'r', 's', 'P', 'Q', 'R', 'S'},                      // 7
+                                       {'*', '+'},                                                         // *
+                                       {'2', 'a', 'b', 'c', 'A', 'B', 'C'},                                // 2
+                                       {'5', 'j', 'k', 'l', 'J', 'K', 'L'},                                // 5
+                                       {'8', 't', 'u', 'v', 'T', 'U', 'V'},                                // 8
+                                       {'0', ' '},                                                         // 0
+                                       {'3', 'd', 'e', 'f', 'D', 'E', 'F'},                                // 3
+                                       {'6', 'm', 'n', 'o', 'M', 'N', 'O'},                                // 6
+                                       {'9', 'w', 'x', 'y', 'z', 'W', 'X', 'Y', 'Z'},                      // 9
+                                       {'#', '@'}};                                                        // #
+
+unsigned char TCA8418LongPressMap[16] = {
+    _TCA8418_ESC,    // C
+    _TCA8418_NONE,   // Navi
+    _TCA8418_NONE,   // Up
+    _TCA8418_NONE,   // Down
+    _TCA8418_NONE,   // 1
+    _TCA8418_LEFT,   // 4
+    _TCA8418_NONE,   // 7
+    _TCA8418_NONE,   // *
+    _TCA8418_UP,     // 2
+    _TCA8418_NONE,   // 5
+    _TCA8418_DOWN,   // 8
+    _TCA8418_NONE,   // 0
+    _TCA8418_NONE,   // 3
+    _TCA8418_RIGHT,  // 6
+    _TCA8418_NONE,   // 9
+    _TCA8418_REBOOT, // #
+};
+
+TCA8418Keyboard::TCA8418Keyboard() : m_wire(nullptr), m_addr(0), readCallback(nullptr), writeCallback(nullptr)
```

Comment:
src/input/TCA8418Keyboard.cpp:153: [medium:warning] Member variable 'TCA8418Keyboard::tap_interval' is not initialized in the constructor. [uninitMemberVar]
src/input/TCA8418Keyboard.cpp:153: [medium:warning] Member variable 'TCA8418Keyboard::backlight_on' is not initialized in the constructor. [uninitMemberVar]